tools/crm_resource_runtime.c

```diff
@@ -965,39 +965,24 @@ generate_resource_params(pe_resource_t *rsc, pe_node_t *node,
 
 bool resource_is_running_on(pe_resource_t *rsc, const char *host)
 {
-    bool found = TRUE;
-    GList *hIter = NULL;
-    GList *hosts = NULL;
+    bool retval = false;
+    GList *lst = NULL;
 
-    if(rsc == NULL) {
-        return FALSE;
-    }
-
-    rsc->fns->location(rsc, &hosts, TRUE);
-    for (hIter = hosts; host != NULL && hIter != NULL; hIter = hIter->next) {
-        pe_node_t *node = (pe_node_t *) hIter->data;
-
-        if(strcmp(host, node->details->uname) == 0) {
-            crm_trace("Resource %s is running on %s\n", rsc->id, host);
-            goto done;
-        } else if(strcmp(host, node->details->id) == 0) {
-            crm_trace("Resource %s is running on %s\n", rsc->id, host);
-            goto done;
-        }
+    if (host == NULL && rsc->running_on == NULL) {
+        crm_trace("Resource %s is not running\n", rsc->id);
+        return false;
     }
 
-    if(host != NULL) {
+    lst = pcmk__singleton_list((gpointer) host);
+    if (pe__rsc_running_on_any_node_in_list(rsc, lst)) {
+        crm_trace("Resource %s is running on %s\n", rsc->id, host);
+        retval = true;
+    } else {
         crm_trace("Resource %s is not running on: %s\n", rsc->id, host);
-        found = FALSE;
-
-    } else if(host == NULL && hosts == NULL) {
-        crm_trace("Resource %s is not running\n", rsc->id);
-        found = FALSE;
```

Comment:
Interestingly the original function returns true if host==NULL and hosts!=NULL (i.e. host==NULL means "is it running anywhere"). Not sure if that's used somewhere

Reply:
This is still different, and the behavior is used in cli_resource_restart()

I don't know if we currently rely on the behavior of pe__rsc_running_on_any_node_in_list() when node_list is NULL; if not, we could maybe repurpose that for this

Reply:
Yes, host can definitely be NULL (at least according to comments).  cli_resource_restart calls resource_is_running_on and passes it a host argument, and the comment for that function says NULL is allowed.

Reply:
It's possible for the node_list to be NULL.  See the last comment in pe__build_node_name_list.  The result of this function can end up as the argument to pe__rsc_running_on_any.